src/dotnet-roslyn-tools/PRTagger/PRTagger.cs

```diff
@@ -2,133 +2,186 @@
+        {
+            gitHubRepoPath = Environment.CurrentDirectory + "\\" + gitHubRepoName;
+            if (!Repository.IsValid(gitHubRepoPath))
             {
-                logger.LogError($"Exception while cloning repo: " + ex);
-                continue;
+                logger.LogInformation("Cloning GitHub repo...");
+                gitHubRepoPath = Repository.Clone(product.RepoHttpBaseUrl, workdirPath: gitHubRepoPath);
             }
-
-            // Find PRs between product commit SHAs
-            var prDescription = new StringBuilder();
-            var isSuccess = PRFinder.PRFinder.FindPRs(previousProductCommitSha, currentProductCommitSha, PRFinder.PRFinder.DefaultFormat, logger, gitHubRepoPath, prDescription);
-            if (isSuccess != 0)
+            else
             {
-                // Error occurred; should be logged in FindPRs method
-                continue;
+                logger.LogInformation($"Repo already exists at {gitHubRepoPath}");
             }
+        }
+        catch (Exception ex)
+        {
+            logger.LogError($"Exception while cloning repo: " + ex);
+            return TagResult.Failed;
+        }
 
-            logger.LogInformation($"Creating issue...");
+        // Find PRs between product commit SHAs
+        var prDescription = new StringBuilder();
+        var isSuccess = PRFinder.PRFinder.FindPRs(previousProductCommitSha, currentProductCommitSha, PRFinder.PRFinder.DefaultFormat, logger, gitHubRepoPath, prDescription);
+        if (isSuccess != 0)
+        {
+            // Error occurred; should be logged in FindPRs method
+            return TagResult.Failed;
+        }
 
-            // Create issue
-            await TryCreateIssueAsync(gitHubRepoName, vsBuild, prDescription.ToString(), settings.GitHubToken, logger).ConfigureAwait(false);
+        var issueTitle = $"[Automated] PRs inserted in VS build {vsBuild}";
+        var hasIssueAlreadyCreated = await HasIssueAlreadyCreatedAsync(gitHubClient, gitHubRepoName, issueTitle, logger).ConfigureAwait(false);
+        if (hasIssueAlreadyCreated)
+        {
+            logger.LogInformation($"Issue with name: {issueTitle} exists in repo: {gitHubRepoName}. Skip creation.");
+            return TagResult.IssueAlreadyCreated;
         }
 
-        return 0;
+        logger.LogInformation($"Creating issue...");
+
+        // Create issue
+        return await TryCreateIssueAsync(gitHubClient, issueTitle, gitHubRepoName, prDescription.ToString(), logger).ConfigureAwait(false);
+    }
+
+    public static async Task<ImmutableArray<(string vsBuild, string vsCommit, string previousVsCommitSha)>> GetVSBuildsAndCommitsAsync(
+        AzDOConnection devdivConnection,
+        ILogger logger,
+        int vsBuildNumber,
+        CancellationToken cancellationToken)
+    {
+        var builds = await devdivConnection.TryGetBuildsAsync(
+            "DD-CB-TestSignVS",
+            logger: logger,
+            maxBuildNumberFetch: vsBuildNumber,
```

Comment:
IMHO 20 might not be enough, especially if we're only running every 12 hours.  And we'll just silently miss builds if it isn't enough.
Is it possible to instead fetch all builds up until the last time we ran (e.g. fetch all builds from the last 12 hours)?

Reply:
I feel each build would takes 1 to 1.5 hour. So if we run it every 12 hours, would it be enough?
It's possible to keep find the last time it gets run, but I am littble bit worry about if
for example: the last insertion issue is created maybe a few months ago and the tagger stops working somehow, later it is fixed, then it will start to generate hundreds of issue when it starts to run

Reply:
OK, how about this, we use both
1. A max number threshold. (Can be specify from the command line)
2. Keep finding the last time we ran. Stop if it is found or the max number is hit.

In this case I feels we should be good?

Reply:
Shame we can't limit our build query to those requestedBy DotNet Bot. But I guess we could miss joint PRs that were created by other teams.

Reply:
So one update on this:
AzDo allows searching for a build, or getting the latest `n` builds.
So there are two ways:
1. Fetch the max number of build, then search the build we need in the list.
2. Find the build by the build number, then compare the how many builds there are since the build has completed.

I use `1` here because I feel it feels more natural to fit the goal 
```
A max number threshold. (Can be specify from the command line)
Keep finding the last time we ran. Stop if it is found or the max number is hit.
```